backends/gdx-backend-lwjgl/src/com/badlogic/gdx/backends/lwjgl/audio/Wav.java

```diff
@@ -90,40 +95,23 @@ public WavInputStream (FileHandle file) {
 
 				// http://www-mmsp.ece.mcgill.ca/Documents/AudioFormats/WAVE/WAVE.html
 				// http://soundfile.sapp.org/doc/WaveFormat/
-				int type = read() & 0xff | (read() & 0xff) << 8;
-				if (type != 1) {
-					String name;
-					switch (type) {
-					case 0x0002:
-						name = "ADPCM";
-						break;
-					case 0x0003:
-						name = "IEEE float";
-						break;
-					case 0x0006:
-						name = "8-bit ITU-T G.711 A-law";
-						break;
-					case 0x0007:
-						name = "8-bit ITU-T G.711 u-law";
-						break;
-					case 0xFFFE:
-						name = "Extensible";
-						break;
-					default:
-						name = "Unknown";
-					}
-					throw new GdxRuntimeException("WAV files must be PCM, unsupported format: " + name + " (" + type + ")");
-				}
+				type = read() & 0xff | (read() & 0xff) << 8;
 
-				channels = read() & 0xff | (read() & 0xff) << 8;
-				if (channels != 1 && channels != 2) throw new GdxRuntimeException("WAV files must have 1 or 2 channels: " + channels);
+				if (type == 0x0055)
+					return; // Handle MP3 in constructor instead
+				else if (type != 0x0001 && type != 0x0003) throw new GdxRuntimeException(
```

Comment:
Dangling `else`. 😛

Reply:
Probably resolved in 879acc1 if I understand what you mean. I'm still not terribly happy with how it's being handled - I think it could be easy to miss the `return` and understand the flow of the code. But all TEt told me was "nice work", so there goes all hope.

Reply:
Yeah, this looks resolved. "Dangling else," where an `else` or `else if` is the next block that would be tried after a certain `return`, `continue`, or `break` probably has minimal to no effect on the performance of the code, but I'd love to be proven wrong. I don't think it affects readability much either... I actually feel like it hurts readability a little to remove the `else`, since there's a clear chain from `if` to `else if` to `else` without needing to look inside any of those blocks, and removing the dangling else forces you to inspect for `return` statements inside the blocks.